File: clients/ios/Lextures/Core/LMS/LMSFeatureModelsAuditLog.swift
import Foundation

/// Admin-console audit event (GET `/api/v1/admin-console/audit-log`).
struct AdminAuditEvent: Decodable, Identifiable, Equatable {
    var eventId: String
    var eventType: String
    var actorId: String
    var timestamp: String
    var orgId: String?
    var targetType: String?
    var targetId: String?

    var id: String { eventId }
}

struct AdminAuditLogResponse: Decodable, Equatable {
    var events: [AdminAuditEvent]
}

File: clients/android/app/src/main/kotlin/com/lextures/android/core/lms/LmsFeatureModelsAuditLog.kt
package com.lextures.android.core.lms

import kotlinx.serialization.Serializable

/** Admin-console audit event (GET `/api/v1/admin-console/audit-log`). */
@Serializable
data class AdminAuditEvent(
    val eventId: String,
    val eventType: String,
    val actorId: String,
    val timestamp: String,
    val orgId: String? = null,
    val targetType: String? = null,
    val targetId: String? = null,
)

@Serializable
data class AdminAuditLogResponse(
    val events: List<AdminAuditEvent> = emptyList(),
)
